### DISCUSSION TITLE
希望添加局部翻译功能

### DISCUSSION THREAD

**Discussion by alvin96** [immersive-translate/immersive-translate#425] [Ideas]
### 新功能 | New Feature

我在用着一个叫 [Mate Translate](https://chrome.google.com/webstore/detail/mate-translate-%E2%80%93-translat/ihmgiclibbndffejedjimfjmfoabpcke) 希望也能添加这个功能
1. 双击选中网站中部分句子
<img width="294" alt="image" src="https://user-images.githubusercontent.com/20142859/235841158-33216cc5-71ac-44c9-ba45-339c99eef861.png">

2. 双击之后会出现小图标 
<img width="214" alt="image" src="https://user-images.githubusercontent.com/20142859/235841314-259d71fc-bd64-4b59-aff7-60ef345f201b.png">

3. 按下小图标就会有译文泡泡出现
<img width="270" alt="image" src="https://user-images.githubusercontent.com/20142859/235841381-28f22127-d905-4fa8-82f6-e1ae79f1382e.png">

由于您的扩展主打沉浸式所以可能可以在按下小图标之后直接让译文显示在主内容区下面（和现在的全局类似，可以让用户在设置中调整样式）
和 #377 类似，希望能划词和双击都实现

### 为什么您想要添加这些功能 | Why you want to add these features

1. 由于我有时候不一定要翻译整个网页更多是用翻译来查找不明白的外语句子所以有了这个功能对我帮助很大
2. 而且双击就能翻译的便捷性远远高于右键再选翻译所以这也能提高用户的效率

### 提议 | Proposal

我是个程序员不过我不会写扩展 如果有什么我能帮的上忙的话我可以试试看

**Comment by jasperyou** [comment]
我之前看别的 issue，这个功能早晚都会做，我也很期待.

这个 issue 中提到的功能，有没有详细的 todo list？我也想参加这个项目

**Reply by alvin96** [reply]
谢谢您的回复，我也对这个功能非常期待
由于我不太熟悉怎么写扩展，不过我的思路是类似这样的

1. 监听用户双击或者选中文字的事件 [Chrome](https://stackoverflow.com/questions/34170032/how-to-get-selected-text-in-chrome-extension-development)，[Firefox](https://stackoverflow.com/questions/71010591/pass-event-listener-from-popup-js-to-background-js-script) ，[其他](https://stackoverflow.com/questions/3545018/selected-text-event-trigger-in-javascript)，[其他](http://jsfiddle.net/qY7gE)
2. 把选中文字翻译
3. 把翻译结果根据设定样式输出到原文底部

希望能帮上忙

**Comment by Leon2xiaowu** [comment]
🎊 这个Feature已经上线啦，扩展商店目前还没审核通过，可以通过手动安装最新的版本进行体验。

**Reply by alvin96** [reply]
谢谢您， 马上就去更新最新版本🎊🎊

**Reply by alvin96** [reply]
我试了一下，效果非常不错我也很喜欢，只是我发现对于下图的元素 (Clollaborator元素) 和非文字的元素翻译不出来的几率有点高，例如按钮里面的文字或者是链接的文字，鼠标hover会显示tooltips的元素也好像会有类似问题，对于这些元素如果不能实现mouse hover的话可能可以让用户手动highlight选取内容然后右键选中translate selected paragraph的选项也不错😊😊
<img width="399" alt="image" src="https://user-images.githubusercontent.com/20142859/236678797-ea176959-859e-43a0-af9e-112be5b2769f.png">

**Reply by Leon2xiaowu** [reply]
这个`Collaborator`应该是长度较短过滤掉了，手动选择内容进行翻译的能力也会考虑的。

**Reply by theowenyoung** [reply]
> 这个`Collaborator`应该是长度较短过滤掉了，手动选择内容进行翻译的能力也会考虑的。

这个可能需要优化一下，可能需要判断下，鼠标滑过强制翻译更符合直觉一点.

**Reply by alvin96** [reply]
> > 这个`Collaborator`应该是长度较短过滤掉了，手动选择内容进行翻译的能力也会考虑的。
> 
> 这个可能需要优化一下，可能需要判断下，鼠标滑过强制翻译更符合直觉一点.

是的，我也觉得鼠标滑过强制翻译更符合直觉一些，不过如果能有把手动选择内容进行翻译的功能作为次要选项也不错，识别不到也可以作为备用方案😊😊
